### ISSUE TITLE
add duplicate_accounts to UserDetails and create api call to append and remove usernames to this field

### ISSUE THREAD

**Issue by jesseallhands** [Couchers-org/couchers#4355]
We need a way to link duplicate accounts (eg. when a user is banned and creates a new account, we need to be able to link them so the various accounts can be found easily again).

At the same time, a related planned feature is to allow users to link their profile with housemates and travel buddies. The idea is that if you live with someone, then you could request to link your profile with theirs and they would show up on each others' profiles as housemates. Similarly, if you are traveling with someone, you can link profiles as travel buddies and this would show up in requests you send. This would also enable the possibility of allowing people to send and receive linked requests. This solves a lot of problems discussed here on the forum: https://community.couchers.org/t/profiles-for-couples-and-friend-groups/431/27

Users could have multiple link types simultaneously housemate, travel_buddy, duplicate_account AND multiple linked accounts (eg. multiple housemates, multiple travel buddies, multiple banned accounts, etc. So I suggest we add housemates, travel_buddies, and duplicate_accounts to UserDetails which would be null by default but have one or more user_id or usernames each.

Then we need an api to update the fields of both user accounts whenever they are linked. 

**For now, we just need to be able to link duplicate accounts of banned users, so the api can be kept very simple, but we should keep the concept of linked accounts in mind when building this on the backend.**

- [ ] add duplicate_accounts to UserDetails
- [x] create API to append and remove APIs
- [ ] update Admin.GetUserDetails so that duplicate accounts are displayed when called (similar to AdminNotes) - though this step may need to be a different issue if aapeli isn't assigned to this issue 😅

**Comment by schradert** [comment]
Moderation strikes me as an important goal for us right now given our discussions so far and active development, so I think linking duplicate accounts of banned users now through something like this could be prioritized. This would be pretty straightforward to implement, I think, pretty much just as you described. @jesseallhands I am curious though what your strategy and workflow currently looks like for identifying and handling banned user accounts and their duplicates. How do you end up finding them? Is there risk of misidentification? Do you just ban the new account as soon as you identify it? Would it be beneficial to somehow flag new accounts for you in a background job that may have similar information to a previously banned account?

Housemates seems like it could follow the same paradigm, but it may benefit from a model similar to what I think is typical and expected for traveling partners on other accommodation booking apps, which is not a permanent association with an account but rather a per-trip designation that allows multiple user accounts to be on the same booking with a shared group chat between everybody.

These two features (housemates and travel partners) seem like they could easily get more complicated. It may suffice to have something much simpler for the time being, like just a link to the others' profiles that users need to be reminded to change to reflect their current travel plans and hosts also ask about to confirm sufficient space. I'm more inclined to develop a complete feature set for these ideas though if others agree it would be beneficial enough.

Curious if @aapeliv has any thoughts about complexity or preferred implementation for any of these use cases.

**Comment by jesseallhands** [comment]
> Moderation strikes me as an important goal for us right now given our discussions so far and active development, so I think linking duplicate accounts of banned users now through something like this could be prioritized. This would be pretty straightforward to implement, I think, pretty much just as you described. 

I agree. As mentioned, the housemates and travel buddy feature is much more complicated, so for now we should just stick to addressing the duplicate account issue. Specifically, creating duplicate_account in UserDetails which can have one or more other usernames and the associated api to append, remove and query usernames to this field. I do think we should use username instead of user_id for the ease of moderators recognizing the linked accounts.

I updated this issue to reflect this.

**Comment by nabramow** [comment]
@schradert are you still working on this?
